### PR TITLE
Update verifiable-randomness-functions and serialize-instruction-data-frontend

### PR DESCRIPTION
**Pull request by EmekaManuel**
### Problem
incorrect url file path for /connecting-to-offchain-data/oracles

### Summary of Changes
changed the file path for /connecting-to-offchain-data/oracles to the correct link.

### CODE REVIEW

File: content/courses/native-onchain-development/serialize-instruction-data-frontend.md [hunk 1]
```diff
@@ -546,7 +575,7 @@ successful.
 
 If you need a bit more time with this project to feel comfortable, have a look
 at the complete
-[solution code](https://github.com/Unboxed-Software/solana-movie-frontend/tree/solution-serialize-instruction-data).
+[solution code](https://github.com/EmekaManuel/movie-review-dapp/tree/solution-serialize-instruction-data).
```
Comment (mikemaccana) [inline content/courses/native-onchain-development/serialize-instruction-data-frontend.md:578]: Update this URL.

File: content/courses/native-onchain-development/serialize-instruction-data-frontend.md [hunk 2]
```diff
@@ -416,8 +435,15 @@ import { FC } from 'react'
 import { Movie } from '../models/Movie'
 import { useState } from 'react'
 import { Box, Button, FormControl, FormLabel, Input, NumberDecrementStepper, NumberIncrementStepper, NumberInput, NumberInputField, NumberInputStepper, Textarea } from '@chakra-ui/react'
```
Comment (mikemaccana) [inline content/courses/native-onchain-development/serialize-instruction-data-frontend.md:437]: I think these imports (and other code) may also need to be updated to reflect the new `movie-review-dapp` you made with `create-solana-dapp`

File: content/courses/native-onchain-development/serialize-instruction-data-frontend.md [hunk 3]
```diff
@@ -281,7 +294,7 @@ The public key of the Solana program we’ll use for this application is
 #### 1. Download the starter code
 
 Before we get started, go ahead and download the
-[starter code](https://github.com/Unboxed-Software/solana-movie-frontend/tree/starter).
+[starter code](https://github.com/EmekaManuel/movie-review-dapp/tree/starter).
```
Comment (mikemaccana) [inline content/courses/native-onchain-development/serialize-instruction-data-frontend.md:297]: Update this URL to point tothe solana-developers fork.

**Review by mikemaccana — CHANGES_REQUESTED** [review]
Just minor fixes left.

**Comment by EmekaManuel** [comment]
> Just minor fixes left.

updated ✅

File: content/courses/native-onchain-development/serialize-instruction-data-frontend.md (hunk 3 above)
Comment (EmekaManuel) [reply content/courses/native-onchain-development/serialize-instruction-data-frontend.md]: > Update this URL to point tothe solana-developers fork.

updated ✅

File: content/courses/native-onchain-development/serialize-instruction-data-frontend.md (hunk 2 above)
Comment (EmekaManuel) [reply content/courses/native-onchain-development/serialize-instruction-data-frontend.md]: fixed!

File: content/courses/native-onchain-development/serialize-instruction-data-frontend.md (hunk 1 above)
Comment (EmekaManuel) [reply content/courses/native-onchain-development/serialize-instruction-data-frontend.md]: updated

File: content/courses/connecting-to-offchain-data/verifiable-randomness-functions.md [hunk 4]
```diff
@@ -1341,13 +1341,18 @@ describe("burry-escrow-vrf", () => {
...
+      // Safely access error message and verify it
+      const errorMessage =
+        error?.error?.errorMessage || error.message || error.toString();
+      console.log(errorMessage);
       assert(
-        e.error.errorMessage ==
+        errorMessage.includes(
           "Current SOL price is not above Escrow unlock price.",
+        ),
+        "Unexpected error message: " + errorMessage,
```
Comment (mikemaccana) [inline content/courses/connecting-to-offchain-data/verifiable-randomness-functions.md:1355]: This is not a problem with your code, just the old code:

`error.error` doesn't make sense. If the first.`error` isn't an error message, we shouldn't be throwing it (JS technically allows us to throw Arrays or Strings or anything else, but that's considered a bad practice). Per CONTRIBUTING.md we can assume code will only throw Errors.

Just get rid of this and use: 

`if error.message.includes()` 

If that doesn't work we have bigger problems!

File: content/courses/native-onchain-development/serialize-instruction-data-frontend.md [hunk 5]
```diff
@@ -225,13 +232,13 @@ equipPlayerSchema.encode(
   buffer,
 );
 
-const instructionBuffer = buffer.slice(0, equipPlayerSchema.getSpan(buffer));
+const instructionBuffer = buffer.subarray(0, equipPlayerSchema.getSpan(buffer));
```
Comment (mikemaccana) [inline content/courses/native-onchain-development/serialize-instruction-data-frontend.md:235]: hah I just learn about subarray. Saving memory. Nice.

**Review by mikemaccana — CHANGES_REQUESTED** [review]
Nearly ready to go. We can get this in today.

File: content/courses/native-onchain-development/serialize-instruction-data-frontend.md (hunk 5 above)
Comment (EmekaManuel) [reply content/courses/native-onchain-development/serialize-instruction-data-frontend.md:235]: > hah I just learn about subarray. Saving memory. Nice.

sure no problem 😂

File: content/courses/connecting-to-offchain-data/verifiable-randomness-functions.md (hunk 4 above)
Comment (EmekaManuel) [reply content/courses/connecting-to-offchain-data/verifiable-randomness-functions.md]: fixed!

File: content/courses/connecting-to-offchain-data/verifiable-randomness-functions.md [hunk 6]
```diff
@@ -1341,13 +1341,24 @@ describe("burry-escrow-vrf", () => {
...
+      if (
+        error.message.includes(
+          "Current SOL price is not above Escrow unlock price.",
+        )
+      ) {
+        console.log("Expected error: " + error.message);
+      } else {
+        console.log("Unexpected error: " + error.message);
+      }
+
```
Comment (mikemaccana) [inline content/courses/connecting-to-offchain-data/verifiable-randomness-functions.md]: You can delete this. Your code below already logs unexpected errors.

**Review by mikemaccana — APPROVED** [review]

**Review by mikemaccana — APPROVED** [review]

**Comment by mikemaccana** [comment]
@EmekaManuel as discussed on Twitter, happy to mark this as the winner for verifiable-randomness-functions as long as the Rust programs still work - will check this later this week.

**Comment by mikemaccana** [comment]
Also marking this as winner for serialize-instruction-data-frontend.md lesson.

**Pull request merged**